### ISSUE TITLE
Set up import process for vendor sources

### ISSUE THREAD

**Issue by RickyDaMa** [googlefonts/googlesans-flex#28]
Writing out the rough import process as exists in Google Sans ([citation](https://github.com/googlefonts/googlesans/blob/35577ff11a979c7c0d772d7cec75f3ae9cb00f43/docs/MAINTAINER.md#steps-in-common-for-designspace--ufo-or-glyphsapp-imports)), complete with check boxes for what is available in this repo on an automated basis:

1. [x] Run the import script on the designspaces, e.g. `python3 scripts/gs-merge-designspace.py --source source/GoogleSans/staging/GoogleSansSomeScript.designspace --target source/GoogleSans/GoogleSans.designspace --import-glyphs-file source/GoogleSans/staging/import_glyphs.txt`
    * Script looks general enough to be useable for Google Sans Flex
    * Locations for staging etc. ought to be standardised to allow for automation. Although, we may be able to do everything in-repo with [git worktree](https://git-scm.com/docs/git-worktree) or another less elegant solution
    * [Script](https://github.com/googlefonts/googlesans/blob/main/scripts/gs-merge-designspace.py)
3. [ ] Extract and merge feature code
    * Difficult to automate but unlikely to be an issue for Jan release due to simple glyphset
    * Instead focus on notifying if the feature code has changed
    * No need to merge until Jan on first release, so can blindly copy for now
2. [ ] Run anchor propogation on glyphs that need it
3. [x] Normalise the designspace
    - Script already exists for this
    - [Script](https://github.com/googlefonts/googlesans/blob/main/scripts/gs-normalize-designspace.py)

**Comment by belluzj** [comment]
@chrissimpkins @MariannaPaszkowska Please find below a proposed workflow for Google Flex, to import regularly the sources from Font Bureau's own GitHub repository. The diagram reads from top to bottom. Feel free to ask any questions, and in any case we can discuss further on Thursday.
![new_situation_ufo_fb](https://user-images.githubusercontent.com/3616772/201713699-6932a61e-b574-47ec-8101-95616c54dcb4.png)

**Comment by belluzj** [comment]
Add to the diagram to notify if feature files need merging by hand. Otherwise ready to implement

**Comment by MariannaPaszkowska** [comment]
Hi @belluzj, we might need to slightly change the plan. Font Bureau will move active development from their repository to their branch in this repository. Any issues that you anticipate?

CC @chrissimpkins @sannorozco

**Comment by belluzj** [comment]
No I don't think so, thanks for checking!

**Comment by RickyDaMa** [comment]
Added a breakdown of the import process to the original issue message, for those that are interested 😃 This will be my main development focus now until it's in a good place

**Comment by madig** [comment]
Regarding production names, these are automatable only when we have a canonical glyph name -> production name mapping. This is the case when designers are using Glyphs.app and use Glyphs.app naming conventions, but Berlow and team have their own naming conventions from what I remember. So it is their job to ensure the correct production names are assigned.

**Comment by dberlow** [comment]
The freestylers will be meeting this week to discuss issues related to this project.  If you could be more specific about how this can be done better, please let us know :)

**Comment by madig** [comment]
Since there's no automatic way to derive production names from custom glyph names, we can't automatically "canonicalize" them on our side. They'll have to match from the start, i.e. you'll have to manually make sure they are what they should be, per glyph, and we import them as-is.

**Comment by MariannaPaszkowska** [comment]
@madig Could you please point us to this glyph name standard? Perhaps, we should review #41 and compare it against this standard.

**Comment by madig** [comment]
It is whatever Glyphs.app says. A copy of the data is available in https://github.com/schriftgestalt/GlyphsInfo/tree/Glyphs3 and included in glyphsLib. An example line:

```xml
<glyph unicode="2078" name="eightsuperior" category="Number" subCategory="Small" case="minor" production="uni2078" description="SUPERSCRIPT EIGHT" />
```

So, one `name` is associated with one `production` name, which we can use in automation.

Edit: it is also possible to maintain a custom file or mapping, but then we need to put in the work to maintain it...

**Comment by dberlow** [comment]
This is a new criteria and will have to be approved Google along with the timeline FB will need for making the changes.

**Comment by chrissimpkins** [comment]
> This is a new criteria and will have to be approved Google along with the timeline FB will need for making the changes.

Reviewing the discussion today. No need to change anything that you are doing at the moment, David. Marianna or I will be in touch if there is a request for a change in your development plan.

**Comment by dberlow** [comment]
﻿Thanks Chris.

We have not been told what glyph’s names are a problem. We suspect DM is referring to additional parts created in our first GS work, and carried into the GSF work, to overcome the use of glyphs app, where its users created a bloated font based on ease of use.

An example of this might be the L-slash we found in sources when we started the GRAD axis, where the /L in the sources would be a composite and the slash a contour. That’s not a “legal” combination within a glyph in the OT spec, and when decomposed to suit the OT spec, leads to that glyph needing /L gvars repeated in the  /L-slash over all the designspace. 

Our alternative approach, as Google saw, allowed GS to gain a GRAD axis with little or no file size gain vs the GS variable we began with and then compressed. That a few glyph parts were added that are outside of the glyphs app database was not a problem for that product. 

For this product we expect file size to again be an issue… so, after examination of the glyphs required for the Latin, Greek, Cyrillic and African extension of Latin we demonstrated this compression starting with the ascii prototype which is approved, and are planning on executing this technique to the limit, with no compromise to design, going forward.

Another issue I have is with the example supplied indicates that Glyphsapp not using names connected to features, for glyphs user by a feature.
